### DISCUSSION TITLE
Solib - Integration with the @walletconnect/sign-client

### DISCUSSION THREAD

**Discussion by daniellnog** [WalletConnect/walletconnect-monorepo#2411] [Developer Support]
Hello, I have a question.

I'm updating my wallet app to WC version 2. Initially, I was using the sign-client api (@walletconnect/sign-client), everything was working well, until try to implement solana. Then, as suggested [here](https://github.com/orgs/WalletConnect/discussions/2324#discussioncomment-5931629), I'm a bit confusing...I deep dive into the [solib documentation](https://solib.dev/) and, the init function looks a bit similar than the SignClient.Init() method. My question now is, should I just use solib init function to connect to a dApp or it should be used just for solana? Because I saw [here](https://solib.dev/#:~:text=while%20WalletConnectConnector%20can%20be%20used%20to%20interact%20with%20all%0A%20%20%20%20//%20wallets%20that%20support%20the%20WalletConnect%20protocol) that the WalletConnectConnector interact with all
wallets that support the WalletConnect protocol.

**Comment by finessevanes** [comment]
We have received and are currently reviewing your discussion for further investigation.

**Comment by devceline** [comment]
https://github.com/WalletConnect/solib/blob/development/example-next/package.json#L11 - For `solib`, you do not need to use sign client, it handles all the signing, transaction sending, etc for you.

**Reply by daniellnog** [reply]
@devceline, thanks for your answer. 

In that case, I will replace my sign-client for your solib, but, in case for ethereum implementation, what should be the methods used to:

eth_signTypedData - ?
eth_sendTransaction - Probably sendTransaction is enough
eth_signTransaction - Probably signTransaction is enough
personal_sign - ?
eth_sign - ?

Thanks.

**Comment by devceline** [comment]
`solib` does not handle ethereum based methods. If your goal is to support both, then your best bet is `@walletconnect/sign-client`, I'll be putting some more thoughts into the original discussion you posted

**Reply by daniellnog** [reply]
Exactly. I need to support both and others, so, in that case, solib should not be ideal for me :(. 

Okay, will wait for your answer there.

Thanks

**Comment by chendinh** [comment]
you should understand that walletconnect is 'middleware tool' between dapp and application. they use the only one rule for code structure and param name. Sonala, bitcoin or polkadot or ether network that have the same general logic and actions (signed message, send transaction, connect, disconnect)
the difference is what you put value in param name or your custom param. Yea! you can custom param for your special case.
-> walletconect can support all network rightnow, dont matter what it is

**Reply by daniellnog** [reply]
Yes, I know walletconnect v2 supports different networks. 

As I've mentioned, using sign-client API, I'm able to work in ethereum network and solana. My original problem is what I've posted [here ](https://github.com/orgs/WalletConnect/discussions/2324)

**Reply by chendinh** [reply]
yep, i developed wallet which support solana network. i hope i can help you soon. Now i have some own business.

**Reply by daniellnog** [reply]
Ok, thanks in advance @chendinh . I hope you can help me :D

**Reply by daniellnog** [reply]
@chendinh is your app open source?

I'm building a isolated app, using the solib.

Once I call connect, it will gives me a wc uri, since I've set qrcode to false. And then, how can I connect with this uri to the Wallet connect?
